tools/src/main/java/org/apache/kafka/trogdor/workload/SustainedConnectionWorker.java

```diff
@@ -0,0 +1,563 @@
+            WorkerUtils.addConfigsToProperties(
+                    this.props, SustainedConnectionWorker.this.spec.commonClientConf(), SustainedConnectionWorker.this.spec.consumerConf());
+        }
+
+        @Override
+        public void refresh() {
+            try {
+
+                if (this.consumer == null) {
+
+                    // Housekeeping to track the number of opened connections.
+                    SustainedConnectionWorker.this.totalConsumerConnections.incrementAndGet();
+
+                    // Create the consumer and fetch the partitions for the specified topic.
+                    this.consumer = new KafkaConsumer<>(this.props, new ByteArrayDeserializer(), new ByteArrayDeserializer());
+                    List<TopicPartition> partitions = this.consumer.partitionsFor(this.topicName).stream()
+                            .map(partitionInfo -> new TopicPartition(partitionInfo.topic(), partitionInfo.partition()))
+                            .collect(Collectors.toList());
+
+                    // Select a random partition and assign it.
+                    this.activePartition = partitions.get(this.rand.nextInt(partitions.size()));
+                    this.consumer.assign(Collections.singletonList(this.activePartition));
+                }
+
+                // The behavior when passing in an empty list is to seek to the end of all subscribed partitions.
+                this.consumer.seekToEnd(Collections.emptyList());
+
+                // Poll to keep the connection alive, ignoring any records returned.
+                this.consumer.poll(Duration.ofMillis(50));
+
+            } catch (Throwable e) {
+
+                // Set the consumer to be recreated on the next cycle.
+                this.closeQuietly();
+
+                // Housekeeping to track the number of opened connections and failed connection attempts.
+                SustainedConnectionWorker.this.totalConsumerConnections.decrementAndGet();
+                SustainedConnectionWorker.this.totalConsumerFailedConnections.incrementAndGet();
+                SustainedConnectionWorker.log.error("Error while refreshing sustained KafkaConsumer connection", e.getMessage());
+
+            }
+
+            // Schedule this again and set to not in use.
+            this.nextUpdate = System.currentTimeMillis() + this.refreshRate;
+            this.inUse = false;
+        }
+
+        @Override
+        protected void closeQuietly() {
+            Utils.closeQuietly(this.consumer, "KafkaConsumer");
+            this.consumer = null;
+            this.activePartition = null;
+        }
+    }
+
+    private static class SustainedConnectionThrottle extends Throttle {
+        SustainedConnectionThrottle(int maxPerPeriod) {
+            super(maxPerPeriod, THROTTLE_PERIOD_MS);
+        }
+    }
+
+    public class MaintainLoop implements Runnable {
+        @Override
+        public void run() {
+            Throttle throttle = new SustainedConnectionThrottle(1);
```

Comment:
What do we want to achieve with this throttle? Do we just want to backoff for `THROTTLE_PERIOD_MS` whenever we can't find a connection you sent? I think we should simply use a `Thread.sleep` call.
To be concrete, I recommend we instantiate a `org.apache.kafka.common.utils.SystemTime` class and use both its `sleep()` to sleep and `milliseconds()` to get the current time

Reply:
Done.